pkg/ebpf/tcp_dns_layer.go

```diff
@@ -0,0 +1,52 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"encoding/binary"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
```

Comment:
nit: can you add `var _ layers.DecodingLayer = &tcpWithDNSSupport{}` near the top of this file to let future readers know what interface is being implemented.